js/packages/openinference-instrumentation-llama-index/test/llamaIndex.test.ts

```diff
@@ -1,8 +1,118 @@
+  instrumentation._modules[0].moduleExports = llamaindex;
+
+  let openAISpy: jest.SpyInstance;
+  let openAITextEmbedSpy: jest.SpyInstance;
+  let openAIQueryEmbedSpy: jest.SpyInstance;
+  beforeAll(() => {
+    instrumentation.enable();
+
+    // Setup the OPENAI_API_KEY
+    process.env["OPENAI_API_KEY"] = "fake-api-key";
+  });
+  afterAll(() => {
+    instrumentation.disable();
+    // Delete the OPENAI_API_KEY
+    delete process.env["OPENAI_API_KEY"];
+  });
+  beforeEach(() => {
+    memoryExporter.reset();
+
+    // Use OpenAI and mock out the calls
+    const response: llamaindex.ChatResponse<llamaindex.ToolCallLLMMessageOptions> =
+      {
+        message: {
+          content: DUMMY_RESPONSE,
+          role: "assistant",
+        },
+        raw: null,
+      };
+    // Mock out the chat completions endpoint
+    openAISpy = jest
+      .spyOn(llamaindex.OpenAI.prototype, "chat")
+      .mockImplementation(() => {
+        return Promise.resolve(response);
+      });
+
+    // Mock out the embeddings response to size 1536 (ada-2)
+    const fakeEmbedding = Array(1536).fill(0);
+    // Mock out the embeddings endpoint
+    openAITextEmbedSpy = jest
+      .spyOn(llamaindex.OpenAIEmbedding.prototype, "getTextEmbeddings")
+      .mockImplementation(() => {
+        return Promise.resolve([fakeEmbedding]);
+      });
+
+    openAIQueryEmbedSpy = jest
+      .spyOn(llamaindex.OpenAIEmbedding.prototype, "getQueryEmbedding")
+      .mockImplementation(() => {
+        return Promise.resolve(fakeEmbedding);
+      });
+  });
+  afterEach(() => {
+    jest.clearAllMocks();
+    openAISpy.mockRestore();
+    openAIQueryEmbedSpy.mockRestore();
+    openAITextEmbedSpy.mockRestore();
+  });
+  it("is patched", () => {
+    expect(isPatched()).toBe(true);
+  });
+  it("should create a span for query engines", async () => {
+    // Create Document object with essay
+    const document = new Document({ text: "lorem ipsum" });
+
+    // Split text and create embeddings. Store them in a VectorStoreIndex
+    const index = await VectorStoreIndex.fromDocuments([document]);
+
+    // Query the index
+    const queryEngine = index.asQueryEngine();
+    const response = await queryEngine.query({
+      query: "What did the author do in college?",
+    });
+
+    // Verify that the OpenAI chat method was called once during synthesis
+    expect(openAISpy).toHaveBeenCalledTimes(1);
+    expect(openAIQueryEmbedSpy).toHaveBeenCalledTimes(1);
+    expect(openAITextEmbedSpy).toHaveBeenCalledTimes(1);
+
+    // Output response
+    expect(response.response).toEqual(DUMMY_RESPONSE);
+
+    const spans = memoryExporter.getFinishedSpans();
+    expect(spans.length).toBeGreaterThan(0);
+
+    // Expect a span for the query engine
+    const queryEngineSpan = spans.find((span) => span.name.includes("Query"));
+    expect(queryEngineSpan).toBeDefined();
```

Comment:
Assuming this will get some more assertions going forward?

Reply:
yeah